equinix/data_source_metal_device.go

```diff
@@ -266,45 +273,49 @@ func dataSourceMetalDeviceRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	if device.HardwareReservation != nil {
-		d.Set("hardware_reservation_id", device.HardwareReservation.ID)
+		// Device schema needs to be updated to include hardware reservation attributes
+		//d.Set("hardware_reservation_id", device.HardwareReservation.ID)
```

Comment:
This may be corrected by https://github.com/equinix/terraform-provider-equinix/pull/291/files#r1126575481